rust/hyperlane-core/src/metrics/agent.rs

```diff
@@ -0,0 +1,23 @@
+use std::time::Duration;
+
+use async_trait::async_trait;
+
+use crate::{ChainResult, U256};
+
+/// Interval for querying the prometheus metrics endpoint.
+/// This should be whatever the prometheus scrape interval is
+pub const METRICS_SCRAPE_INTERVAL: Duration = Duration::from_secs(60);
+
+/// Trait to be implemented by all chain-specific agent implementations,
+/// to support gathering agent metrics.
+#[async_trait]
+pub trait AgentMetricsFetcher: Send + Sync {
+    /// Fetch the balance of the wallet address associated with the chain provider.
+    async fn get_balance(&self, address: String) -> ChainResult<U256>;
```

Comment:
Curious why String vs H256 was preferable. Strings feel unnecessarily more prone to errors and it's harder to understand what format they'll be in. I think needing to change the formatting of the ethereum signer_address to accommodate this is a good example of the fragility of using strings. Elsewhere in the codebase we use H256 as the chain-agnostic representation for addresses when the value isn't intended for human reading & needs to be formatted a certain way

Iiuc using a String results in needing to parse from String -> H160 for Ethereum which feels less nice than H256 -> H160. And for cosmos we can convert from H256 -> the address easily with `CosmosAddress::from_h256(address, &conf.get_prefix()`

Reply:
As discussed, will stick with `String` because when the Cosmos H256 digest is reinterpreted as an address, all 32 bytes are considered (instead of just the last 20 bytes), because contract addresses are 32 bytes long. Using strings leaves it up to the bech32 encoding function to figure out whether the string is an EOA or contract address